### ISSUE TITLE
Mocking does not actually mock

### ISSUE THREAD

**Issue by davemasterdave** [Codeception/AspectMock#42]
I added the configuration as suggested in the docs for Yii 1.1 and when I run the unit tests with PHPUnit 3.7, the methods I am mocking do not get overridden.

This is my bootstrap.php for PHPUnit:
```php
<?php

$yiit=dirname(__FILE__).'/../../../yii/framework/yiit.php';
if (!file_exists($yiit))
    $yiit = dirname(__FILE__).'/../../vendor/yiisoft/yii/framework/yiit.php';
$config=dirname(__FILE__).'/../config/test.php';

$kernel = AspectMock\Kernel::getInstance();
$kernel->init([
    'debug' => true,
    'includePaths' => [
        __DIR__.'/../',
    ],
    'excludePaths' => [
        __DIR__.'/../runtime',
        __DIR__.'/../data',
        __DIR__.'/../commands',
        __DIR__.'/../bin',
        __DIR__.'/../migrations',
    ],
]);
$kernel->loadFile($yiit);

Yii::createWebApplication($config);

?>
```
```php
//The basic class I am trying to mock:
class FounderBadgeType implements BadgeTypeInterface {

    public function getBadgeOrder() {
        return 1;
    }
    //...
}
```
```php
//Unit Test
$type = test::double(new FounderBadgeType(), ['getBadgeOrder' => 5]);
$this->assertEquals(5,$type->getBadgeOrder()); //FAILS? Still returns 1, not 5
```

**Comment by qml** [comment]
@davemasterdave you solved this problem? I have the same problem.

**Comment by davemasterdave** [comment]
@qml to be honest, I just ended up not using this library because of this issue. I forgot why I closed this issue...

**Comment by JoeVieira** [comment]
@DavertMik any comments on this issue? I run into the same issues. 
* Mocks don't seem to mock.
* Examples in readme.md don't seem to work.

**Comment by JoeVieira** [comment]
Thanks for responding, so in general you see these types of issues when the autoloader isn't functioning correctly?

**Comment by DavertMik** [comment]
Yes. AspectMock (and Go AOP) need to get access to autoloader function and patch it in runtime. So it will replace all `require` and `include` in order to load patched files

**Comment by kenjis** [comment]
Why do you pass [yiit.php](https://github.com/yiisoft/yii/blob/master/framework/yiit.php)?
It seems there is no autoloader in it.

**Comment by JoeVieira** [comment]
@DavertMik Thanks for clarifying. Cheers.

**Comment by DavertMik** [comment]
> Why do you pass yiit.php?

It's not yiit...
Sorry, I can't help further as I didn't work with Yii for a long time. Just copied the example from a project I was working on.

**Comment by kenjis** [comment]
@DavertMik Sorry, I asked @JoeVieira .

At least, [YiiBase.php](https://github.com/yiisoft/yii/blob/master/framework/YiiBase.php#L878) has an autoloader. So you need to pass it to `$kernel->loadFile()`.

But Yii has a method to [registers a new class autoloader](https://github.com/yiisoft/yii/blob/master/framework/YiiBase.php#L630). I don't know how to handle it with AspectMock.

**Comment by JoeVieira** [comment]
I don't use Yii. @DavertMik was using that as an example.

**Comment by kenjis** [comment]
Oh, sorry.
